### ISSUE TITLE
Black background of PNG images with transparent background in the Wysiwyg editor

### ISSUE THREAD

**Issue by boesbo** [OpenMage/magento-lts#3998]
### Preconditions (*)
1. OpenMage 2.7.0

### Steps to reproduce (*)
1. Open the Static Block creation page
2. Go to the editor and load an image. A PNG with a transparent background

### Expected result (*)
1. The thumbnail created is with black background instead of transparent
2. Thumbnail viewable on edit is with black background instead of transparent

### Actual result (*)
1. The problem only affects thumbnail creation and preview display in the editor.
2. Regarding Thumb creation I am testing a fix that involves adding `$image->keepTransparency(true);` to line `386` of the file `app/code/core/Mage/Cms/Model/Wysiwyg/Images/Storage.php`
3. As for displaying the preview in the editor, go to Controller: `app/code/core/Mage/Adminhtml/controllers/Cms/WysiwygController.php` Action: `directiveAction()`. I still haven't found a way to solve it.

I attach images of my GD version.

#### Thumb `app/code/core/Mage/Cms/Model/Wysiwyg/Images/Storage.php::resizeFile`
![Screen Shot 2024-05-16 at 16 17 21 PM](https://github.com/OpenMage/magento-lts/assets/12744378/85913c94-3248-4275-8290-98a048f6fc73)

#### Preview in editor `app/code/core/Mage/Adminhtml/controllers/Cms/WysiwygController.php::directiveAction`
![Screen Shot 2024-05-16 at 16 17 31 PM](https://github.com/OpenMage/magento-lts/assets/12744378/aba6a56c-a813-4251-ac34-72b507fd1c32)

![Screen Shot 2024-05-16 at 15 33 36 PM](https://github.com/OpenMage/magento-lts/assets/12744378/30127d1f-94fb-4f75-848c-3ad546712a3e)

**Comment by boesbo** [comment]
> I'm testing this but I don't think it's the editor's fault because I see the black background already right after uploading it
> 
> <img alt="Screenshot 2024-05-17 alle 10 07 53" width="1093" src="https://private-user-images.githubusercontent.com/909743/331514305-e1a5fde8-e15b-417f-a0e1-21231c2fc26a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU5MzM2NzgsIm5iZiI6MTcxNTkzMzM3OCwicGF0aCI6Ii85MDk3NDMvMzMxNTE0MzA1LWUxYTVmZGU4LWUxNWItNDE3Zi1hMGUxLTIxMjMxYzJmYzI2YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxN1QwODA5MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iOGM3NWI2M2I2ZDI5Zjg1MTJhMTNiZjZkNDBkZTgyMTgzMjEzMjI0MzYwNzZkN2U4NDljN2M4MzQ4OTc2YjY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ExLaFk0nRM_GM_KYU4b2GfWSAsR9hfiiUX3ajqQUpDw">
> at that moment I don't think the new wyswyg has nothing to do with it

Yes, the problem is in the two files I mentioned above

**Comment by fballiano** [comment]
same thing happening with the old editor on latest v19

<img width="1054" alt="Screenshot 2024-05-17 alle 10 12 12" src="https://github.com/OpenMage/magento-lts/assets/909743/c5d6a429-317c-410b-9543-75afd7af527e">

**Comment by boesbo** [comment]
> same thing happening with the old editor on latest v19
> 
> <img alt="Screenshot 2024-05-17 alle 10 12 12" width="1054" src="https://private-user-images.githubusercontent.com/909743/331515542-c5d6a429-317c-410b-9543-75afd7af527e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU5MzM4NjcsIm5iZiI6MTcxNTkzMzU2NywicGF0aCI6Ii85MDk3NDMvMzMxNTE1NTQyLWM1ZDZhNDI5LTMxN2MtNDEwYi05NTQzLTc1YWZkN2FmNTI3ZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxN1QwODEyNDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYmY3NTc5ZjMwMTZiMzBjZmM5MmVkYmU3ZDAxZWExN2Y3NDUyNTJlYzdlNGZjODFjZjAxZTJhYmI1YjQ3ZDI3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.EHBLCHXAkITUuspCBzlVXIVh3-8NG9pBMXZP_22SX8k">

Yes, by selling the history I noticed that it is something that we have been carrying with me I imagine since Magento 1.9. Do you think this is intentional behavior? It seems strange.

**Comment by fballiano** [comment]
mmmm I really don't know :-\

all images are always re-saved in order to remove any possible form of "virus" within the image, but to remove transparency I've no idea why

**Comment by pquerner** [comment]
Its imagemagick. Where it gets its settings tho, that I don't know.
I remember something from the TYPO3 days: https://stackoverflow.com/questions/18809692/why-do-imagemagick-based-thumbnail-images-of-pdf-files-in-typo3-have-black-backg https://forge.typo3.org/issues/37060

I dont know if that setting is available to customize from within magento, or if its only sitting in some php.ini file. I know too little about this.

**Comment by boesbo** [comment]
> Its imagemagick. Where it gets its settings tho, that I don't know. I remember something from the TYPO3 days: https://stackoverflow.com/questions/18809692/why-do-imagemagick-based-thumbnail-images-of-pdf-files-in-typo3-have-black-backg https://forge.typo3.org/issues/37060
> 
> I dont know if that setting is available to customize from within magento, or if its only sitting in some php.ini file. I know too little about this.

That part of the script uses the GD library not ImageMagick:
 `$image = Varien_Image_Adapter::factory('GD2');`

**Comment by pquerner** [comment]
Well this perhaps then
https://stackoverflow.com/questions/5913284/php-gd-transparent-areas-goes-black

(im checking out here, im just googleing stuff anyway)

**Comment by boesbo** [comment]
Yes, this is it. In the two files I wrote at the beginning, transparency management is missing. But it's so obvious that it makes me doubt. It doesn't seem like an oversight.

I can also act by creating a PR, but first I want to better understand if this behavior has a purpose, since it is only visible in the editor.

**Comment by pquerner** [comment]
Well, all this system comes from a time where it wasnt the norm and browsers had troubles with transparent files? I think up to IE11 no IE browser learned to do it or something? So it could be very well just that. :D

**Comment by boesbo** [comment]
In the next few days I will propose a patch. I am very clear on the problem. Transparency is simply not managed in the two phases: 1. thumb creation, 2. editor preview url
